### ISSUE TITLE
Sanitizing systemd timer service

### ISSUE THREAD

**Issue by LecrisUT** [smallstep/docs#65]
I have followed the instructions [on the systemd timer certificate renewal](https://smallstep.com/docs/step-ca/certificate-authority-server-production#renewal-using-systemd-timers) and everything works great but, there are some errors/warnings that pop-up.
- `Failed to get unit file state for {INSTANCE_NAME}.service: No such file or directory`
- `cert-renewer@{INSTANCE_NAME}.service: Control process exited, code=exited status=1`

The latter issue is because of `ExecStartPre`, and although the comments and documentation say that `ExecCondition` requires `systemd` version of >= 243, my CentOS version `239 (239-45.el8)` runs it just fine. Maybe the default can be `ExecCondition` and have the user change it if needed?

For the former one I've used a variation of the `lighttpd` instruction, and I'm greeted with that message whenever it executes. Any suggestions on how sanitize the log for that issue?

**Comment by tashian** [comment]
Hi @LecrisUT,

Hmm, the first one is most likely coming from the `ExecStartPost` in `cert-renewer@.service`, which tries to reload or restart a service with the same name as the variable you called `{INSTANCE_NAME}` above. The goal is to have it silently continue if that service doesn't exist.

Here's the latest version of it:

```
; Try to reload or restart the service that parallels this cert-renewer, if it is enabled
; if the service doesn't exist, forge ahead.
ExecStartPost=/usr/bin/env bash -c "if ! systemctl --quiet is-enabled %i.service ; then exit 0; fi; systemctl try-rel
oad-or-restart %i"
```

(I recently updated this line to avoid the error you got. Could you check your `cert-renewer@.service` and confirm that it matches the above snippet?)

The second log message is normal.

I'm glad you're using `ExecCondition`, it's the preferred option. But, for now we will keep it as `ExecStartPost` for the broadest compatibility. I tested a couple of distros manually in the past, and I definitely ran into some widely-adopted distros that do not (yet) have `ExecCondition` support.

**Comment by LecrisUT** [comment]
Interesting, so the override confs on `ExecStartPost` and `ExecStartPre` are added to rather than replaced? The error for that message also appears early in the log, before the main executable's if i remember correctly.

**Comment by tashian** [comment]
That's correct, and you can have as many `ExecStartPre` and `ExecStartPost` as you want.
